test/test_client_db_hooks.js

```diff
@@ -0,0 +1,70 @@
+import app           from './server';
+import db            from '../src/models/db';
+import {
+  datastreams,
+  DatastreamsEntity,
+  entities,
+  featuresOfInterest,
+  FeaturesOfInterestEntity,
+  historicalLocations,
+  HistoricalLocationsEntity,
+  locations,
+  LocationsEntity,
+  observations,
+  ObservationsEntity,
+  observedProperties,
+  ObservedPropertiesEntity,
+  sensors,
+  SensorsEntity,
+  things,
+  ThingsEntity
+} from './constants';
+import supertest     from 'supertest';
+
+const server           = supertest.agent(app.listen(8890));
+
+db().then(models => {
+  describe('db hooks', () => {
+    beforeEach(done => {
+      models.sequelize.transaction(transaction => {
+        return Promise.all(Object.keys(entities).map(name => {
+          return models[name].destroy({ transaction, where: {} });
+        }));
+      }).then(() => done());
```

Comment:
I know we have this pattern all over the code, but `beforeEach` takes a `Promise` as well. So you can get rid of `done` and do

```javascript
beforeEach(() => {
  return models.sequelize.transaction(...);
});
```